azure/functions/decorators/core.py

```diff
@@ -0,0 +1,138 @@
+    be present on the request in order to invoke the function. """
+    FUNCTION = "function"
+    """A function-specific API key is required. This is the default value if
+    none is provided. """
+    ANONYMOUS = "anonymous"
+    """No API key is required."""
+    ADMIN = "admin"
+    """The master key is required."""
+
+
+class Cardinality(StringifyEnum):
+    """Used for all non-C# languages. Set to many in order to enable
+    batching. If omitted or set to one, a single message is passed to the
+    function. """
+    ONE = "one"
+    """Singe message passed to the function."""
+    MANY = "many"
+    """Multiple messaged passed to the function per invocation."""
+
+
+class AccessRights(StringifyEnum):
+    """Access rights for the connection string. The default is manage,
+    which indicates that the connection has the Manage permission. """
+    MANAGE = "manage"
+    """Confers the right to manage the topology of the namespace, including
+    creating and deleting entities. """
+    LISTEN = "listen"
+    """Confers the right to listen (relay) or receive (queue, subscriptions)
+    and all related message handling. """
+
+
+class Binding(ABC):
+    """Abstract binding class which captures common attributes and
+    functions. :meth:`get_dict_repr` can auto generate the function.json for
+    every binding, the only restriction is please ensure __init__ parameter
+    names of any binding class are snake case form of corresponding
+    attribute in function.json when new binding classes are created.
+    Ref: https://aka.ms/azure-function-binding-http """
+
+    @staticmethod
+    @abstractmethod
+    def get_binding_name() -> str:
+        pass
+
+    def __init__(self, name: str,
+                 direction: BindingDirection,
+                 is_trigger: bool,
+                 data_type: Optional[DataType] = None):
+        self.type = self.get_binding_name()
+        self.is_trigger = is_trigger
+        self.name = name
+        self._direction = direction
+        self._data_type = data_type
+        self._dict = {
+            "direction": self._direction,
+            "dataType": self._data_type,
+            "type": self.type
+        }
+
+    @property
+    def data_type(self) -> Optional[int]:
+        if self._data_type is None:
+            return self._data_type
+        return self._data_type.value
+
+    @property
+    def direction(self) -> int:
+        return self._direction.value
+
+    def get_dict_repr(self) -> Dict:
+        #
+        for p in getattr(self, 'init_params', []):
+            if p not in ['data_type', 'self']:
+                self._dict[camel_case(p)] = getattr(self, p, None)
+
+        return self._dict
+
+    def get_binding_json(self) -> str:
+        return json.dumps(self.get_dict_repr(), cls=CustomJsonEncoder)
+
+    def __str__(self):
+        return self.get_binding_json()
+
+
+class Trigger(Binding, ABC, metaclass=ABCBuildDictMeta):
```

Comment:
Add the following link to a PyDoc comment for all the three `Trigger`, `InputBinding` and `OutputBinding` - https://aka.ms/functions-triggers-bindings-overview

Just future-proofing for reference.